tikv-client/src/main/java/com/pingcap/tikv/TiSessionCache.java

```diff
@@ -0,0 +1,25 @@
+package com.pingcap.tikv;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TiSessionCache {
+  private static Map<String, TiSession> sessionCachedMap = new ConcurrentHashMap<>();
+
+  // Since we create session as singleton now, configuration change will not
+  // reflect change
+  public static TiSession getSession(TiConfiguration conf) {
```

Comment:
do we need to synchronize?

Reply:
Yes. Just added and change `ConcurentHashMap` to `HashMap`